### ISSUE TITLE
Random persistent queue corruption on system hard reset

### ISSUE THREAD

**Issue by prakashguru** [elastic/logstash#12040]
- Version: 7.6.1
- Operating System: Windows 10 Enterprise N - 10.0.17134
- Config File (if you have sensitive info, please remove it):
path.data: D:\my_app\logstash\data
config.reload.automatic: true
config.reload.interval: 3s
queue.type: persisted 
queue.max_bytes: 4gb
path.logs: D:\my_app\logstash\logs

- Steps to Reproduce:
The host windows operating system corrupted dumped multiple times (5-6) times in a week due to some driver issues.  

The windows vm able to start/recover from crash without much issues. But logstash failed to restart on two occasions with the error "**Expected page version=2 but found version=0**"

```
[2020-06-21T07:53:28,016][ERROR][org.logstash.execution.AbstractPipelineExt] Logstash failed to create queue.
org.logstash.ackedqueue.io.MmapPageIOV2$PageIOInvalidVersionException: Expected page version=2 but found version=0
	at org.logstash.ackedqueue.io.MmapPageIOV2.validateVersion(MmapPageIOV2.java:370) ~[logstash-core.jar:?]
	at org.logstash.ackedqueue.io.MmapPageIOV2.recover(MmapPageIOV2.java:137) ~[logstash-core.jar:?]
	at org.logstash.ackedqueue.Queue.open(Queue.java:210) ~[logstash-core.jar:?]
	at org.logstash.ackedqueue.ext.JRubyAckedQueueExt.open(JRubyAckedQueueExt.java:100) ~[logstash-core.jar:?]
	at org.logstash.ackedqueue.ext.JRubyWrappedAckedQueueExt.initialize(JRubyWrappedAckedQueueExt.java:42) ~[logstash-core.jar:?]
	at org.logstash.ackedqueue.QueueFactoryExt.create(QueueFactoryExt.java:39) ~[logstash-core.jar:?]
	at org.logstash.execution.AbstractPipelineExt.openQueue(AbstractPipelineExt.java:176) [logstash-core.jar:?]
	at org.logstash.execution.AbstractPipelineExt$INVOKER$i$0$0$openQueue.call(AbstractPipelineExt$INVOKER$i$0$0$openQueue.gen) [jruby-complete-9.2.9.0.jar:?]
	at org.jruby.internal.runtime.methods.JavaMethod$JavaMethodN.call(JavaMethod.java:831) [jruby-complete-9.2.9.0.jar:?]
	at org.jruby.ir.targets.InvokeSite.invoke(InvokeSite.java:183) [jruby-complete-9.2.9.0.jar:?]
	at D_3a_.Syskey.Scribbler.logstash_minus_app.logstash_minus_core.lib.logstash.java_pipeline.RUBY$method$initialize$0(D:/logstash-app/logstash-core/lib/logstash/java_pipeline.rb:28) [jruby-complete-9.2.9.0.jar:?]
	at D_3a_.Syskey.Scribbler.logstash_minus_app.logstash_minus_core.lib.logstash.pipeline_action.create.RUBY$method$execute$0(D:/logstash-app/logstash-core/lib/logstash/pipeline_action/create.rb:36) [jruby-complete-9.2.9.0.jar:?]
	at D_3a_.Syskey.Scribbler.logstash_minus_app.logstash_minus_core.lib.logstash.pipeline_action.create.RUBY$method$execute$0$__VARARGS__(D:/logstash-app/logstash-core/lib/logstash/pipeline_action/create.rb) [jruby-complete-9.2.9.0.jar:?]
	at org.jruby.internal.runtime.methods.CompiledIRMethod.call(CompiledIRMethod.java:84) [jruby-complete-9.2.9.0.jar:?]
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:70) [jruby-complete-9.2.9.0.jar:?]
	at org.jruby.ir.targets.InvokeSite.invoke(InvokeSite.java:183) [jruby-complete-9.2.9.0.jar:?]
	at D_3a_.Syskey.Scribbler.logstash_minus_app.logstash_minus_core.lib.logstash.agent.RUBY$block$converge_state$2(D:/logstash-app/logstash-core/lib/logstash/agent.rb:326) [jruby-complete-9.2.9.0.jar:?]
	at org.jruby.runtime.CompiledIRBlockBody.callDirect(CompiledIRBlockBody.java:136) [jruby-complete-9.2.9.0.jar:?]
	at org.jruby.runtime.IRBlockBody.call(IRBlockBody.java:77) [jruby-complete-9.2.9.0.jar:?]
	at org.jruby.runtime.IRBlockBody.call(IRBlockBody.java:71) [jruby-complete-9.2.9.0.jar:?]
	at org.jruby.runtime.Block.call(Block.java:125) [jruby-complete-9.2.9.0.jar:?]
	at org.jruby.RubyProc.call(RubyProc.java:274) [jruby-complete-9.2.9.0.jar:?]
	at org.jruby.internal.runtime.RubyRunnable.run(RubyRunnable.java:105) [jruby-complete-9.2.9.0.jar:?]
```
Did an investigation, The PQ file "page.7" is a zero filled 64 MB file. The "MmapPageIOV2.validateVersion" expects the version in the first byte as 2 but it was 0. 

No idea why that file is zero filled. 

Is there anyway for logstash recover from this failure gracefully?

**Comment by yaauie** [comment]
Possible fix #12554 

Recovery: with logstash shut down, delete the all-zeroes page file and then run `bin/pqrepair`.

**Comment by yaauie** [comment]
Fix has been merged and will be a part of the 7.11 release. 

> Recovery: with logstash shut down, delete the all-zeroes page file and then run `bin/pqrepair`.
